v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DataStreamToSpanner.java

```diff
@@ -643,20 +667,77 @@ public static PipelineResult run(Options options) {
                     .setMaxRpcTimeout(org.threeten.bp.Duration.ofMinutes(4))
                     .setMaxAttempts(1)
                     .build());
+
+    // Validate shadow table Spanner config - both instance and database must be specified together
+    String shadowTableSpannerInstanceId = options.getShadowTableSpannerInstanceId();
```

Comment:
This whole flow of creating an alternate shadow table spanner config etc, can probably be moved to a different method.